### PR TITLE
LPS-169909 DuplicateFolderNameException when adding a folder with same name

### PR DESCRIPTION
**Pull request by locpham97**
Hi team,
Please help me review this ticket.
This is the PR for fixing https://issues.liferay.com/browse/LPS-169909
Thanks,
Loc
cc:@thienquach87

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/servlet/RepositoryBrowserServlet.java [hunk 1]
```diff
@@ -224,6 +226,12 @@ protected void doPut(
 			_sendResponse(httpServletResponse, HttpServletResponse.SC_OK);
 		}
 		catch (PortalException portalException) {
+			if (portalException instanceof DuplicateFolderNameException) {
+				_sendError(httpServletResponse, portalException.getMessage());
+
+				return;
+			}
+
```
Comment (adolfopa) [inline modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/servlet/RepositoryBrowserServlet.java:234]: Can you split this in two distinct catch clauses?
```
catch (DuplicateFolderNameException duplicateFolderNameException) {
   _sendError(...);
}
catch  (PortalException portalException) {
   ...
}
```

**Review by adolfopa — CHANGES_REQUESTED** [review]

**Comment by thienquach87** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/servlet/RepositoryBrowserServlet.java (hunk 1 above)
Comment (thienquach87) [reply modules/apps/document-library/document-library-taglib/src/main/java/com/liferay/document/library/taglib/internal/servlet/RepositoryBrowserServlet.java]: @adolfopa 
I updated source, thank you!

**Comment by thienquach87** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
@thienquach87 See alternative fix in https://github.com/liferay-lima/liferay-portal/pull/3845. I've fixed an additional case (duplicate file entry). Also, to keep things consistent, I've used the same error reporting mechanisms already used by the servlet (return status + session message).

**Comment by thienquach87** [comment]
> @thienquach87 See alternative fix in #3845. I've fixed an additional case (duplicate file entry). Also, to keep things consistent, I've used the same error reporting mechanisms already used by the servlet (return status + session message).

I got your idea, your fix will work in the other expectation.
Thank you!

**Pull request closed without merging**